solutions/usaco-570.mdx

```diff
@@ -2,72 +2,372 @@
 id: usaco-570
 source: USACO Silver 2015 December
 title: Switching on the Lights
-author: Óscar Garries
+author: Akshar Yeccherla
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_lightson_silver_dec15.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_lightson_silver_dec15.html)
 
-<LanguageSection>
-<CPPSection>
+## Explanation
+
+We can traverse through the rooms that are lit and connected to the room $(1,1)$ using floodfill, since we are essentially traversing a connected component. We can then turn on all the switches in each room we visit.
+
+However, we run into an issue when a newly lit room that becomes part of the main connected component can't be visited since floodfill already visited the lit rooms neighboring it. 
```

Comment:
```suggestion
However, we run into an issue when a newly lit room that becomes part of the main connected component can't be visited since the floodfill already visited the lit rooms neighboring it. 
```